code/modules/paperwork/paperbin.dm

```diff
@@ -31,48 +31,59 @@
 	return
 
 /obj/item/weapon/paper_bin/attack_hand(mob/user as mob)
-	if(ishuman(user))
-		var/mob/living/carbon/human/H = user
-		var/obj/item/organ/external/temp = H.organs_by_name[BP_R_ARM]
-		if (H.hand)
-			temp = H.organs_by_name[BP_L_ARM]
-		if(temp && !temp.is_usable())
-			user << SPAN_NOTICE("You try to move your [temp.name], but cannot!")
-			return
-	var/response = ""
-	if(!papers.len > 0)
-		response = alert(user, "Do you take regular paper, or Carbon copy paper?", "Paper type request", "Regular", "Carbon-Copy", "Cancel")
-		if (response != "Regular" && response != "Carbon-Copy")
-			add_fingerprint(user)
-			return
-	if(amount >= 1)
-		amount--
-		if(amount==0)
-			update_icon()
-
-		var/obj/item/weapon/paper/P
-		if(papers.len > 0)	//If there's any custom paper on the stack, use that instead of creating a new paper.
-			P = papers[papers.len]
-			papers.Remove(P)
+	if (istype(loc, /turf))
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			var/obj/item/organ/external/temp = H.organs_by_name[BP_R_ARM]
+			if (H.hand)
+				temp = H.organs_by_name[BP_L_ARM]
+			if(temp && !temp.is_usable())
+				user << SPAN_NOTICE("You try to move your [temp.name], but cannot!")
+				return
+		var/response = ""
+		if(!papers.len > 0)
+			response = alert(user, "Do you take regular paper, or Carbon copy paper?", "Paper type request", "Regular", "Carbon-Copy", "Cancel")
+			if (response != "Regular" && response != "Carbon-Copy")
+				add_fingerprint(user)
+				return
+		if(amount >= 1)
+			amount--
+			if(amount==0)
+				update_icon()
+
+			var/obj/item/weapon/paper/P
+			if(papers.len > 0)	//If there's any custom paper on the stack, use that instead of creating a new paper.
+				P = papers[papers.len]
+				papers.Remove(P)
+			else
+				if(response == "Regular")
+					P = new /obj/item/weapon/paper
+					if(Holiday == "April Fool's Day")
+						if(prob(30))
+							P.info = "<font face=\"[P.crayonfont]\" color=\"red\"><b>HONK HONK HONK HONK HONK HONK HONK<br>HOOOOOOOOOOOOOOOOOOOOOONK<br>APRIL FOOLS</b></font>"
+							P.rigged = 1
+							P.updateinfolinks()
+				else if (response == "Carbon-Copy")
+					P = new /obj/item/weapon/paper/carbon
+
+			P.loc = user.loc
+			user.put_in_hands(P)
```

Comment:
`P.loc = user.loc` already handles in `put_in_hands(...)` with proper calls like `forceMove(...)`